applications/sdp/mspi/src/main.c

```diff
@@ -35,6 +39,8 @@
 #define VEVIF_IRQN(vevif)   VEVIF_IRQN_1(vevif)
 #define VEVIF_IRQN_1(vevif) VPRCLIC_##vevif##_IRQn
 
+#define NRF_GPIOHSPADCTRL ((NRF_GPIOHSPADCTRL_Type *)NRF_P2_S_BASE)
```

Comment:
```suggestion
#ifdef CONFIG_SOC_NRF54L15
#define NRF_GPIOHSPADCTRL ((NRF_GPIOHSPADCTRL_Type *)NRF_P2_S_BASE)
#else
#error "Unsupported SoC for SDP MSPI"
#endif
```

I think that such defines should be surrounded by soc specific #if's. Or concatenate it from symbols defined for specific SOC. 
Also above.